js/models/messages.js

```diff
@@ -991,6 +991,49 @@
       });
     },
 
+    /**
+     * Uploads attachments, previews and quotes.
+     * If body is too long then it is also converted to an attachment.
+     *
+     * @returns The uploaded data which includes: body, attachments, preview and quote.
+     */
+    async uploadData() {
+      // TODO: In the future it might be best if we cache the upload results if possible.
+      // This way we don't upload duplicated data.
```

Comment:
we will probably need this for the `resend` and `retrysend` feature too

Reply:
`resend` and `retrysend` use this function, so by doing it here any functions will call this will benefit